pkg/endpoint/endpoint.go

```diff
@@ -909,8 +909,9 @@ func parseEndpoint(ctx context.Context, owner regeneration.Owner, policyGetter p
 
 	ep.initDNSHistoryTrigger()
 
-	// Validate the options that were parsed
-	ep.SetDefaultOpts(ep.Options)
+	// Set default options, unsupported options were already dropped by
+	// ep.Options.UnmarshalJSON
+	ep.SetDefaultOpts(nil)
```

Comment:
nit: typo in the commit message `SetDeafaultOpts()` --> `SetDefaultOpts()`

Reply:
I'll let this slide :-)